Source/BoundaryConditions/WarpXFieldBoundaries.cpp

```diff
@@ -36,9 +51,13 @@ void WarpX::ApplyBfieldBoundary (const int lev, PatchType patch_type, DtType a_d
 {
     if (PEC::isAnyBoundaryPEC()) {
         if (patch_type == PatchType::fine) {
-            PEC::ApplyPECtoBfield( Bfield_fp[lev], lev, patch_type);
+            PEC::ApplyPECtoBfield( { get_pointer_Bfield_fp(lev, 0),
+                                     get_pointer_Bfield_fp(lev, 1),
+                                     get_pointer_Bfield_fp(lev, 2) }, lev, patch_type);
         } else {
-            PEC::ApplyPECtoBfield( Bfield_cp[lev], lev, patch_type);
+            PEC::ApplyPECtoBfield( { get_pointer_Bfield_cp(lev, 0),
+                                     get_pointer_Bfield_cp(lev, 1),
+                                     get_pointer_Bfield_cp(lev, 2) }, lev, patch_type);
         }
```

Comment:
I guess we will want to implement similar changes here for PMC in case we have split components for B in the PML, correct? Would make sense to do this in a separate PR of course, just making a note here.

Reply:
yes - and PMC boundary condition is not supported. Will add this to the Boundary condition project.